### ISSUE TITLE
implement deephit loss

### ISSUE THREAD

**Issue by peterchristofferholm** [peterchristofferholm/discotime#4]

**Comment by peterchristofferholm** [comment]
We need the following functions:

```
def deephit_loss(
    logits: torch.Tensor, time: torch.Tensor, event: torch.Tensor
) -> torch.Tensor:

    ...
```

```
def deephit_predict(
    logits: torch.Tensor, discretization_grid: torch.Tensor, timepoints: torch.Tensor, 
) -> torch.Tensor:

    ....
```

**Comment by daplaci** [comment]
deephit_loss is done, but requires some testing due to the differences in how the different risks are considered in the original implementation (still alive is not one of the risk)
